packages/matter.js/src/codec/BtpCodec.ts

```diff
@@ -0,0 +1,144 @@
+/**
+ * @license
+ * Copyright 2022-2023 Project CHIP Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { ByteArray, Endian } from "../util/ByteArray.js";
+import { DataReader } from "../util/DataReader.js";
+import { DataWriter } from "../util/DataWriter.js";
+
+export interface BtpHandshakeRequest {
+    versions: number[],
+    attMtu: number,
+    clientWindowSize: number,
+}
+
+export interface BtpHandshakeResponse {
+    version: number,
+    attMtu: number,
+    windowSize: number
+}
+
+export interface BtpPacketPayload {
+    ackNumber?: number,
+    sequenceNumber: number,
+    msgLength?: number,
+    segmentPayload?: ByteArray
+}
+
+export interface BtpHeader {
+    handshakeBit: boolean,
+    managementBit: boolean,
+    ackMsgBit: boolean,
+    endSegmentBit: boolean,
+    beginSegmentBit: boolean
+}
+
+export interface BtpPacket {
+    header: BtpHeader,
+    payload: BtpPacketPayload
+}
+
+export const enum BtpHeaderBits {
+    HandshakeBit = 0b01000000,
+    ManagementMsg = 0b00100000,
+    AckMsg = 0b00001000,
+    EndSegment = 0b00000100,
+    BeginSegment = 0b00000001
+}
+
+export const enum BtpOpcode {
+    HandshakeManagementOpcode = 0x6c,
+}
+
+const HANDSHAKE_HEADER = 0b01100101;
+
+export class BtpCodec {
+
+    static decodeBtpHandshakeRequest(data: ByteArray): BtpHandshakeRequest {
+        const reader = new DataReader(data, Endian.Little);
+        return this.decodeRequestPayload(reader);
+    }
+
+    static decodeBtpPacket(data: ByteArray): BtpPacket {
+        const reader = new DataReader(data, Endian.Little);
+
+        return {
+            header: this.decodeHeader(reader),
+            payload: this.decodeBtpPacketPayload(reader)
+        };
+
+    }
+
+    static encodeBtpHandshakeResponse({ version, attMtu, windowSize }: BtpHandshakeResponse): ByteArray {
+
+        const writer = new DataWriter(Endian.Little);
+        writer.writeUInt8(HANDSHAKE_HEADER);
+        writer.writeUInt8(BtpOpcode.HandshakeManagementOpcode);
+        writer.writeUInt8(version & 0x0f); //reserved bit and final version
+        writer.writeUInt16(attMtu);
+        writer.writeUInt8(windowSize);
+        return writer.toByteArray();
+    }
+
+    private static decodeBtpPacketPayload(reader: DataReader<Endian.Little>): BtpPacketPayload {
+
+        const ackNumber = reader.readUInt8();
+        const sequenceNumber = reader.readUInt8();
+        const msgLength = reader.readUInt16();
+        const segmentPayload = reader.getRemainingBytes();
+
+        return { ackNumber, sequenceNumber, msgLength, segmentPayload };
+    }
+
+    private static decodeRequestPayload(reader: DataReader<Endian.Little>): BtpHandshakeRequest {
+
+        const ver = [];
```

Comment:
now it is a Array of "any" ... we know that it will be number so number array would be better :-)

Reply:
And ideally move it down to the place where you use it.